sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -35,19 +35,15 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        AtomicReference<Throwable> cause = new AtomicReference<>();
-        return Flux.fromIterable(credentials)
-            .flatMap(p -> p.getToken(request).onErrorResume(t -> {
-                if (cause.get() != null) {
-                    t.initCause(cause.get());
-                }
-                cause.set(t);
-                return Mono.empty();
-            }))
+        final StringBuilder errorMsg = new StringBuilder();
+        return Flux.fromIterable(credentials).flatMap(p -> p.getToken(request).onErrorResume(t -> {
+            if (t.getMessage() != null && !t.getMessage().contains("authentication unavailable")) {
+                throw new RuntimeException(UnavailableError+p.getClass().getSimpleName()+"authentication failed.",t);
```

Comment:
This error message is not properly formatted, spacing issues will show up before 'authentication failed' text.